### PR TITLE
add a mapping from atom type to element

### PR DESCRIPTION
**Pull request by BBBuZHIDAO**
There is a function in LAMMPS called "**map_element2type**". It can build a mapping from atom type defined in data file to element type in potential file. So we just need to copy this mapping to make the setting more flexible. To be more standardized, the model filename would be read in pair_coeff setting.
One more thing, I set the flag "one_coeff“ to 1, which means the pair_coeff only allows * * call. But it doesn't affect the use.

### CODE REVIEW

**Comment by brucefan1983** [comment]
This is a great improvement. Could you please update the readme file?

File: interface/lammps/USER-NEP/pair_NEP.cpp [hunk 1]
```diff
@@ -60,6 +60,7 @@ PairNEP::~PairNEP()
   if (allocated) {
     memory->destroy(setflag);
     memory->destroy(cutsq);
+    delete[] type_map;
```
Comment (brucefan1983) [inline interface/lammps/USER-NEP/pair_NEP.cpp:63]: There is no `delete[] map`?

File: src/nep.cpp [hunk 2]
```diff
@@ -2776,6 +2780,35 @@ void NEP3::init_from_file(const std::string& potential_filename, const bool is_r
...
+    for (n = 0; n < paramb.num_types; ++n) {
+      if (element_name == element_list[n]) {
+        type_map[itype] = n;
+        break;
+      }
+    }
+
+    // check if no corresponding element
+    if (n == paramb.num_types) {
+      std::cout << "There is no element " << element_name << " in the potential file." << std::endl;
```
Comment (brucefan1983) [inline src/nep.cpp:2806]: Is this function only called by a single MPI thread?

Comment (brucefan1983) [reply src/nep.cpp:2806]: I mean it is better to confine the error reporting to a single thread using `if (is_rank_0)`.

File: interface/lammps/USER-NEP/pair_NEP.cpp [hunk 3]
```diff
@@ -103,6 +114,7 @@ void PairNEP::init_style()
 
   bool is_rank_0 = (comm->me == 0);
   nep_model.init_from_file(model_filename, is_rank_0);
+  nep_model.update_type_map(atom->ntypes, type_map, elements);
```
Comment (brucefan1983) [inline interface/lammps/USER-NEP/pair_NEP.cpp:117]: you can pass `is_rank_0` into `update_type_map()`.

File: interface/lammps/USER-NEP/pair_NEP.cpp (hunk 1 above)
Comment (BBBuZHIDAO) [reply interface/lammps/USER-NEP/pair_NEP.cpp:63]: Pair::map will be deleted by Pair::~Pair()
image

File: src/nep.cpp (hunk 2 above)
Comment (BBBuZHIDAO) [reply src/nep.cpp:2806]: Yep, you are right, it's better. I check the previous error reporting method and find there is no `if (is_rank_0)`.
image

Comment (BBBuZHIDAO) [reply src/nep.cpp:2806]: But why we don't use LAMMPS_NS::error to report? (make LAMMPS to consider the MPI hhhh)

Comment (brucefan1983) [reply src/nep.cpp:2806]: We cannot use LAMMPS stuff in `nep.cpp` as it is also used for other packages such as `calorine`.

Comment (brucefan1983) [reply src/nep.cpp:2806]: It is ok to report error for all threads. This is not a very important thing.

File: interface/lammps/USER-NEP/pair_NEP.cpp (hunk 1 above)
Comment (brucefan1983) [reply interface/lammps/USER-NEP/pair_NEP.cpp:63]: OK

**Pull request merged**